src/Symfony/Component/Form/Extension/Core/Type/IntegerType.php

```diff
@@ -36,9 +37,19 @@ public function buildForm(FormBuilderInterface $builder, array $options)
      */
     public function configureOptions(OptionsResolver $resolver)
     {
+        $scale = function (Options $options) {
+            if (null !== $options['precision']) {
+                trigger_error('The form option "precision" is deprecated since version 2.7 and will be removed in 3.0. Use "scale" instead.', E_USER_DEPRECATED);
+            }
+
+            return $options['precision'];
+        };
+
         $resolver->setDefaults(array(
-            // default precision is locale specific (usually around 3)
+            // deprecated as of Symfony 2.7, to be removed in Symfony 3.0.
             'precision' => null,
+            // default scale is locale specific (usually around 3)
+            'scale' => null,
```

Comment:
the default scale should not be ``null`` but the closure you defined above reading the precision option for BC